worker/provisioner/provisioner_task.go

```diff
@@ -695,11 +695,22 @@ func constructStartInstanceParams(
 		StatusCallback:    machine.SetInstanceStatus,
 	}
 
-	// ToDo ProvisionerParallelization 2017-10-15
-	// Call PreviewAvailabilityZoneSuccess()
+	if zonedEnv, ok := task.broker.(providercommon.ZonedEnviron); ok {
+		zone, err := zonedEnv.DeriveAvailabilityZone(startInstanceParams)
+		switch {
+		case err != nil && !errors.IsNotImplemented(err):
```

Comment:
I believe you got rid of the NotImplementedf implementation, so no need to check for it any more. just `if err != nil ...`

Reply:
true, thank you, updated